packages/ui/src/common/providers/modal/provider.tsx

```diff
@@ -25,7 +26,12 @@ export const ModalContextProvider = (props: Props) => {
       }
     },
     hideModal: () => {
-      if (isClosing || !MODAL_WITH_CLOSE_CONFIRMATION.includes((modal ?? '') as ModalNames)) {
+      const [state] = currentModalMachine ?? []
+      if (
+        isClosing ||
+        !MODAL_WITH_CLOSE_CONFIRMATION.includes((modal ?? '') as ModalNames) ||
+        state?.matches('success')
+      ) {
```

Comment:
Which is better readable?
```suggestion
      if (isClosing || !MODAL_WITH_CLOSE_CONFIRMATION.includes((modal ?? '') as ModalNames) || state?.matches('success')) {
```